src/Main.java

```diff
@@ -1,5 +1,346 @@
+        {
+            System.out.println("Calculator result is " + totalValueAdvanced + " .");
+        }
+
+        //Task 1g
+        System.out.println("You are on task 1g \n");
+
+        int numberForOperationBonus;
+        int initialIntegerValue = 0;
+        String operationToPerformBonus;
+        int totalValueBonus = 0;
+        System.out.println("Is initial integer value defined: y/n");
+        String isIntegerValueDefined = scanner.next();
+
+        if (isIntegerValueDefined.contains("n")) {
+            System.out.println("Please enter the initial integer value:");
+            initialIntegerValue = scanner.nextInt();
+            totalValueBonus = initialIntegerValue;
+        }
+
+        // Could also be done with boolean true or false
+
+        totalValueBonus = initialIntegerValue;
+
+        do {
+            int previousValueOfTheSum = totalValueBonus;
+            System.out.println("Which operation would you like to use: *,/,+,-,=?");
+            operationToPerformBonus = scanner.next();
+
+            if (operationToPerformBonus.equals("+")) {
+                numberForOperationBonus = enterInteger();
+                totalValueBonus = addingIntegers(previousValueOfTheSum, numberForOperationBonus);
+                System.out.println("The sum of " + previousValueOfTheSum + " and " + numberForOperationBonus + " is " + totalValueBonus);
+            } else if (operationToPerformBonus.equals("-")) {
+                numberForOperationBonus = enterInteger();
+                totalValueBonus = subtractingIntegers(previousValueOfTheSum, numberForOperationBonus);
+                System.out.println("The difference of " + previousValueOfTheSum + " and " + numberForOperationBonus + " is " + totalValueBonus);
+            } else if (operationToPerformBonus.equals("*")) {
+                numberForOperationBonus = enterInteger();
+                totalValueBonus = multiplyingIntegers(previousValueOfTheSum, numberForOperationBonus);
+                System.out.println("The product of " + previousValueOfTheSum + " and " + numberForOperationBonus + " is " + totalValueBonus);
+            } else if (operationToPerformBonus.equals("/")) {
+                numberForOperationBonus = enterInteger();
+                totalValueBonus = divisioningIntegers(previousValueOfTheSum, totalValueBonus);
+                System.out.println("The quotient of " + previousValueOfTheSum + " and " + numberForOperationBonus + " is " + totalValueBonus);
+            }
+
+        } while (!operationToPerformBonus.equals("="));
+        {
+            System.out.println("Calculator result is " + totalValueBonus + " .");
+        }
+
+        //Zadatak 2
+        System.out.println("You are on task 2 \n");
+
+        int index = 0;
+        int addNumberToArray;
+        int[] arrayOfNumbers = new int[15];
+
+        for (int i = 0; i <= arrayOfNumbers.length; i++) {
+            addNumberToArray = enterInteger();
+
```

Comment:
ok solution but could we break this into logical wholes within multiple functions?

Reply:
We could probably. Will check some ideas.